substrate/frame/support/src/traits/misc.rs

```diff
@@ -37,8 +37,6 @@ pub const DEFENSIVE_OP_PUBLIC_ERROR: &str = "a defensive failure has been trigge
 pub const DEFENSIVE_OP_INTERNAL_ERROR: &str = "Defensive failure has been triggered!";
 
 /// Trait to get the number of variants in any enum.
-///
-/// NOTE: can be removed once <https://doc.rust-lang.org/std/mem/fn.variant_count.html> is stable.
```

Comment:
well, I think because that [nightly feature](https://doc.rust-lang.org/std/mem/fn.variant_count.html) won't for our result runtime enum `RuntimeHoldReason` / `RuntimeFreezeReason`, e.g.: check this gist: https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=57f1091d103a488db9290f637d292e20 .

We need real composite count (sum of inner enum variants), e.g. if we want to remove `MaxHolds` / `MaxFreezes` as [discussed bellow](https://github.com/paritytech/polkadot-sdk/pull/2657#discussion_r1444428053).

I was even considering renaming the `trait VariantCount` to `trait CompositeVariantCount."

Or maybe even better, leave `VariantCount` with that `NOTE` and use it only for inner enums with `#[pallet::composite_enum]`, so when nighly becomes stable we could remove `VariantCount`,
and create `trait CompositeVariantCount` dedicated just to the result runtime enum like `RuntimeHoldReason` / `RuntimeFreezeReason`.

So at the end, we will have just `CompositeVariantCount`.

@bkchr  wdyt?